### PR TITLE
chore: remove ajv schema verification of modern configuration

### PR DESCRIPTION
**Pull request by zllkjc**
In Modern.js, running Ajv causes every CLI command to take an extra 50~150ms of time. 

We found that the vast of users do not rely on schema verification, it can accurately filter configuration errors through type definition.

So we decide to remove Ajv, depends only on `.ts` config file.

### <samp>🤖 Generated by Copilot at 4417a73</samp>

Removed the `Ajv`-based config validation feature from the core and the plugins of the `modern.js` CLI. This feature was unnecessary and complicated the codebase. Deleted the files and code related to the config validation schemas, the `validateSchema` hooks, and the `repeatKeyWarning` utility. Updated the function calls and types to reflect the changes.

### <samp>🤖 Generated by Copilot at 4417a73</samp>

*  Simplified the config validation logic by removing the use of `Ajv` and the schema-related code and arguments from the core and the plugins ([link](https://github.com/web-infra-dev/modern.js/pull/4911/files?diff=unified&w=0#diff-e27238819be6ceb9b03acb191d10d84e2cb6c622203b96e78e6d1800360dfd40L1-R2), [link](https://github.com/web-infra-dev/modern.js/pull/4911/files?diff=unified&w=0#diff-e27238819be6ceb9b03acb191d10d84e2cb6c622203b96e78e6d1800360dfd40L20-R14), [link](https://github.com/web-infra-dev/modern.js/pull/4911/files?diff=unified&w=0#diff-e39dd2579a14a04a7dc0e7aa69e8f5db1187eb68e486a051674bf66ebef85953L122-R123), [link](https://github.com/web-infra-dev/modern.js/pull/4911/files?diff=unified&w=0#diff-90bfe33a6d04494133cef1df1aecaafcd95c0b671b2c57cdbd7f94558cbe2048L22), [link](https://github.com/web-infra-dev/modern.js/pull/4911/files?diff=unified&w=0#diff-f472a5ce0e7134f1da83593d2dcb0dd9ddc59172c73af45de3ffe3235b45cc64L1), [link](https://github.com/web-infra-dev/modern.js/pull/4911/files?diff=unified&w=0#diff-f472a5ce0e7134f1da83593d2dcb0dd9ddc59172c73af45de3ffe3235b45cc64L20), [link](https://github.com/web-infra-dev/modern.js/pull/4911/files?diff=unified&w=0#diff-8da1726a5da82e9c6dcf2604766e45cd2616c55121f32411e510fc6ba3e2d947L30), [link](https://github.com/web-infra-dev/modern.js/pull/4911/files?diff=unified&w=0#diff-4b1f5602071332c4d6082b05fd2f7e36ef649f6df244fb5bfcd6d680bfb5695bL1-L2), [link](https://github.com/web-infra-dev/modern.js/pull/4911/files?diff=unified&w=0#diff-4b1f5602071332c4d6082b05fd2f7e36ef649f6df244fb5bfcd6d680bfb5695bL9-L13), [link](https://github.com/web-infra-dev/modern.js/pull/4911/files?diff=unified&w=0#diff-932edaaba6ef6ad3ee82c5785fb560ab2408c7a74dbcb62086114bcb7d13af67L5), [link](https://github.com/web-infra-dev/modern.js/pull/4911/files?diff=unified&w=0#diff-ddb13338fa5e4286f0affca3a481539c4265f010c94db3bb4beb2c5ba9202033L23-L40), [link](https://github.com/web-infra-dev/modern.js/pull/4911/files?diff=unified&w=0#diff-849678029fc6a0220011a6b65bb584a44818279e956362f98c9c652fab4da014L17-L24), [link](https://github.com/web-infra-dev/modern.js/pull/4911/files?diff=unified&w=0#diff-d9a53b8123181f7fd750ff419573b15ba8cda6f7d9fd484895ae733dcfbadee7L12-L20), [link](https://github.com/web-infra-dev/modern.js/pull/4911/files?diff=unified&w=0#diff-57f91437065c96e8eedcf6d66ca05e6f3041f9124de48cc3b0f8dc65c8e437f5L5-L12), [link](https://github.com/web-infra-dev/modern.js/pull/4911/files?diff=unified&w=0#diff-c5b135a5f97484fcf0d659c37b3fa54a1f8220e568e47785925656fcb231f402L28-L41), [link](https://github.com/web-infra-dev/modern.js/pull/4911/files?diff=unified&w=0#diff-f346ff1e7d78405c3b1be5cb417d1a6f70a475a88ae4fccdcd40a51090c6b376L18-L25), [link](https://github.com/web-infra-dev/modern.js/pull/4911/files?diff=unified&w=0#diff-a7c5f07b8103f54cc7910493cc4598a797a04f0cd3f50d351753d1e86ce5742dL65-L74), [link](https://github.com/web-infra-dev/modern.js/pull/4911/files?diff=unified&w=0#diff-ac8e4d04c278607aa550bd4bcb05c9ca6918677f96b4dddd925f495e246c74f9L105-L123), [link](https://github.com/web-infra-dev/modern.js/pull/4911/files?diff=unified&w=0#diff-52a9ec1272d09f100d1e982ff44a4d1314d9baf1bab819e72f6c197d570a6d25), [link](https://github.com/web-infra-dev/modern.js/pull/4911/files?diff=unified&w=0#diff-e9f1ecf9741ea57e963e521cab3a83af78ee7a8d44e081170caf5db0932e8bba), [link](https://github.com/web-infra-dev/modern.js/pull/4911/files?diff=unified&w=0#diff-c175dcf07d8bba0817983a5771f41bd97820527c43a56097b80a3df652455129), [link](https://github.com/web-infra-dev/modern.js/pull/4911/files?diff=unified&w=0#diff-e14e0f71ea9a5d0be15865ad60526c510d4b553bad0e4b9a6727e41670ab29b3), [link](https://github.com/web-infra-dev/modern.js/pull/4911/files?diff=unified&w=0#diff-b8c359e3c9e2d5064ae5676f2008217cce4f67a798766bc8182ce17158ace768))
* Updated the `createResolveConfig` function call in `createCli.ts` to match the new signature of the function ([link](https://github.com/web-infra-dev/modern.js/pull/4911/files?diff=unified&w=0#diff-e39dd2579a14a04a7dc0e7aa69e8f5db1187eb68e486a051674bf66ebef85953L122-R123))
* Removed the `validateSchema` workflow from the `manager` object in `manager.ts` and the `BaseHooks` type in `hooks.ts` ([link](https://github.com/web-infra-dev/modern.js/pull/4911/files?diff=unified&w=0#diff-90bfe33a6d04494133cef1df1aecaafcd95c0b671b2c57cdbd7f94558cbe2048L22), [link](https://github.com/web-infra-dev/modern.js/pull/4911/files?diff=unified&w=0#diff-8da1726a5da82e9c6dcf2604766e45cd2616c55121f32411e510fc6ba3e2d947L30))
* Removed the `onSchemaError` property from the `CoreOptions` interface in `coreOptions.ts` ([link](https://github.com/web-infra-dev/modern.js/pull/4911/files?diff=unified&w=0#diff-f472a5ce0e7134f1da83593d2dcb0dd9ddc59172c73af45de3ffe3235b45cc64L1), [link](https://github.com/web-infra-dev/modern.js/pull/4911/files?diff=unified&w=0#diff-f472a5ce0e7134f1da83593d2dcb0dd9ddc59172c73af45de3ffe3235b45cc64L20))
* Removed the `PluginValidateSchema` interface from `index.ts` ([link](https://github.com/web-infra-dev/modern.js/pull/4911/files?diff=unified&w=0#diff-4b1f5602071332c4d6082b05fd2f7e36ef649f6df244fb5bfcd6d680bfb5695bL1-L2), [link](https://github.com/web-infra-dev/modern.js/pull/4911/files?diff=unified&w=0#diff-4b1f5602071332c4d6082b05fd2f7e36ef649f6df244fb5bfcd6d680bfb5695bL9-L13))
* Removed the `repeatKeyWarning` export from `utils/index.ts` ([link](https://github.com/web-infra-dev/modern.js/pull/4911/files?diff=unified&w=0#diff-932edaaba6ef6ad3ee82c5785fb560ab2408c7a74dbcb62086114bcb7d13af67L5))
* Removed the `validateSchema` hook from the `bff`, `proxy`, `ssg`, `swc`, `tailwind`, and `designToken` plugins in their respective files ([link](https://github.com/web-infra-dev/modern.js/pull/4911/files?diff=unified&w=0#diff-ddb13338fa5e4286f0affca3a481539c4265f010c94db3bb4beb2c5ba9202033L23-L40), [link](https://github.com/web-infra-dev/modern.js/pull/4911/files?diff=unified&w=0#diff-d9a53b8123181f7fd750ff419573b15ba8cda6f7d9fd484895ae733dcfbadee7L12-L20), [link](https://github.com/web-infra-dev/modern.js/pull/4911/files?diff=unified&w=0#diff-c5b135a5f97484fcf0d659c37b3fa54a1f8220e568e47785925656fcb231f402L28-L41), [link](https://github.com/web-infra-dev/modern.js/pull/4911/files?diff=unified&w=0#diff-f346ff1e7d78405c3b1be5cb417d1a6f70a475a88ae4fccdcd40a51090c6b376L18-L25), [link](https://github.com/web-infra-dev/modern.js/pull/4911/files?diff=unified&w=0#diff-a7c5f07b8103f54cc7910493cc4598a797a04f0cd3f50d351753d1e86ce5742dL65-L74), [link](https://github.com/web-infra-dev/modern.js/pull/4911/files?diff=unified&w=0#diff-ac8e4d04c278607aa550bd4bcb05c9ca6918677f96b4dddd925f495e246c74f9L105-L123))
* Removed the test cases for the `validateSchema` hook from the `bff`, `proxy`, and `designToken` plugins in their respective files ([link](https://github.com/web-infra-dev/modern.js/pull/4911/files?diff=unified&w=0#diff-849678029fc6a0220011a6b65bb584a44818279e956362f98c9c652fab4da014L17-L24), [link](https://github.com/web-infra-dev/modern.js/pull/4911/files?diff=unified&w=0#diff-57f91437065c96e8eedcf6d66ca05e6f3041f9124de48cc3b0f8dc65c8e437f5L5-L12), [link](https://github.com/web-infra-dev/modern.js/pull/4911/files?diff=unified&w=0#diff-ac8e4d04c278607aa550bd4bcb05c9ca6918677f96b4dddd925f495e246c74f9L105-L123))
* Deleted the files `traverseSchema.ts`, `patchSchema.ts`, `repeatKeyWarning.ts`, `repeatKeyWarning.test.ts`, and `schema.test.ts` as they are no longer needed ([link](https://github.com/web-infra-dev/modern.js/pull/4911/files?diff=unified&w=0#diff-52a9ec1272d09f100d1e982ff44a4d1314d9baf1bab819e72f6c197d570a6d25), [link](https://github.com/web-infra-dev/modern.js/pull/4911/files?diff=unified&w=0#diff-e9f1ecf9741ea57e963e521cab3a83af78ee7a8d44e081170caf5db0932e8bba), [link](https://github.com/web-infra-dev/modern.js/pull/4911/files?diff=unified&w=0#diff-c175dcf07d8bba0817983a5771f41bd97820527c43a56097b80a3df652455129), [link](https://github.com/web-infra-dev/modern.js/pull/4911/files?diff=unified&w=0#diff-e14e0f71ea9a5d0be15865ad60526c510d4b553bad0e4b9a6727e41670ab29b3), [link](https://github.com/web-infra-dev/modern.js/pull/4911/files?diff=unified&w=0#diff-b8c359e3c9e2d5064ae5676f2008217cce4f67a798766bc8182ce17158ace768))

## Checklist

- [ ] I have added changeset via `pnpm run change`.
- [ ] I have updated the documentation.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

File: packages/toolkit/utils/package.json [hunk 1]
```diff
@@ -50,7 +50,6 @@
       "import": "./dist/esm/universal/pluginDagSort.js",
       "default": "./dist/cjs/universal/pluginDagSort.js"
     },
-    "./ajv": "./compiled/ajv/index.js",
```
Comment (chenjiahan) [inline packages/toolkit/utils/package.json:53]: We can delete the `compiled/ajv` and `compiled/ajv-keywords` folder now

Comment (chenjiahan) [reply packages/toolkit/utils/package.json:53]: And` better-ajv-errors`

Comment (zllkjc) [reply packages/toolkit/utils/package.json:53]: done!

**Review by chenjiahan — APPROVED** [review]

**Comment by chenjiahan** [comment]
👏

**Pull request merged**